Improve state module docs, replace references to state.highstate/state.sls with state.apply

terminalmage opened the pull request:
Resolves #32037.

terminalmage committed:
Improve state module docs, replace references to state.highstate/state.sls with state.apply

Resolves #32037.

terminalmage commented:
Would like some critique on this from one or more of the following: @cachedout, @thatch45, @UtahDave.

cachedout commented on doc/topics/tutorials/states_pt1.rst:
```diff
@@ -125,13 +125,16 @@ Next, let's run the state we created. Open a terminal on the master and run:
+modules listed for it will be downloaded, compiled, and executed.
+
+.. note::
+    This action is referred to as a "highstate".
```
I think we need to expand this documentation to include an explanation stating that `state.apply` is what was previously known as `state.highstate` and has been aliased to make it easier to understand.

cachedout commented:
I think we just need to add an explanation that we're transitioning to the new language, both in the docs and perhaps in the release notes as well.

terminalmage committed:
Clarify prior role of state.highstate in states tutorial

terminalmage committed:
Move highstate usage details to top of state.apply docstring

terminalmage commented:
@cachedout Good point, I adjusted the docs per your recommendations in https://github.com/saltstack/salt/pull/32038/commits/74ee8c5

I also moved the highstate usage details above the state.sls usage details in the ``state.apply`` docstring. The thinking here being that people hitting the docs for ``state.apply`` via the docs will more often than not be doing so in the context of a highstate action, so making them scroll down to read docs relevant to this doesn't seem like a good user experience. Finally, I referenced the "highstate" section of the States Tutorial from the ``state.apply`` docstring.

terminalmage committed:
Add reference to state tutorial to state.apply docstring
